pkg/kdump/kdump-view.jsx

```diff
@@ -55,130 +58,177 @@ const _ = cockpit.gettext;
+                }
+            },
+            _internal: {
+                ...settings._internal
+            }
+        };
 
-                    <FormGroup fieldId="kdump-settings-ssh-directory" label={_("Directory")}>
-                        <TextInput id="kdump-settings-ssh-directory" key="directory"
-                                   placeholder="/var/crash" value={directory}
-                                   data-stored={directory}
-                                   onChange={value => this.props.onChange("path", value)} />
-                    </FormGroup>
-                </>
-            );
+        if (storageLocation === "ssh") {
+            newSettings.targets.ssh.server = server;
+            newSettings.targets.ssh.sshkey = sshkey;
         }
 
-        const targetDescription = {
-            local: _("Local filesystem"),
-            nfs: _("Remote over NFS"),
-            ssh: _("Remote over SSH"),
-        };
-        // we don't support all known storage options currently
-        const storageDest = this.state.storeDest;
-        return (
+        if (storageLocation === "nfs") {
+            newSettings.targets.nfs.server = server;
+            newSettings.targets.nfs.export = exportPath;
+        }
+
+        handleSave(newSettings)
+                .then(Dialogs.close)
+                .finally(() => setIsSaving(false))
+                .catch(error => {
+                    if (error.details) {
+                        // avoid bad summary like "systemd job RestartUnit ["kdump.service","replace"] failed with result failed"
+                        // if we have a more concrete journal
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error.details))));
+                    } else {
+                        // without a journal, show the error as-is
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error))));
+                    }
+                });
+    };
+
+    return (
+        <Modal position="top" variant="small" id="kdump-settings-dialog" isOpen
+               title={_("Crash dump location")}
+               onClose={Dialogs.close}
+               footer={
+                   <>
+                       <Button variant="primary"
+                               isLoading={isSaving}
+                               isDisabled={isSaving}
+                               onClick={saveSettings}>
+                           {_("Save changes")}
+                       </Button>
+                       <Button variant="link"
+                               isDisabled={isSaving}
+                               className="cancel"
+                               onClick={Dialogs.close}>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+               }>
+            {error && <ModalError dialogError={typeof error == 'string' ? error : error.message} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18435-20230309-102332-3351cf2f-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)